metrics-runtime/src/common.rs

```diff
@@ -1,58 +1,68 @@
 use crate::data::AtomicWindowedHistogram;
+use metrics_core::{IntoKey, Key, ScopedString};
 use metrics_util::StreamingIntegers;
 use quanta::Clock;
-use std::borrow::Cow;
 use std::ops::Deref;
 use std::sync::atomic::{AtomicI64, AtomicU64, Ordering};
 use std::sync::Arc;
 use std::time::{Duration, Instant};
 
-/// Optimized metric name.
-///
-/// This can either be a [`&'static str`](str) or [`String`].
-pub type MetricName = Cow<'static, str>;
-
 /// A scope, or context, for a metric.
 ///
 /// Not interacted with directly by end users, and only exposed due to a lack of trait method
 /// visbility controls.
 ///
 /// See also: [Sink::scoped](crate::Sink::scoped).
-#[derive(PartialEq, Eq, Hash, Clone)]
-pub enum MetricScope {
+#[derive(PartialEq, Eq, Hash, Clone, Debug)]
+pub enum Scope {
     /// Root scope.
     Root,
 
     /// A nested scope, with arbitrarily deep nesting.
     Nested(Vec<String>),
 }
 
-impl MetricScope {
-    pub(crate) fn into_scoped(self, name: MetricName) -> String {
+impl Scope {
+    pub(crate) fn into_scoped(self, name: ScopedString) -> ScopedString {
```

Comment:
The naming doesnt really follow the rust conventions (https://rust-lang-nursery.github.io/api-guidelines/naming.html#c-conv). I think you would want a special consuming constructor for scoped string instead.

Reply:
The problem is that `ScopedString` is actually just a type alias for `Cow<'static, str>`, so the best we could do, I think, is an extension trait, which feels like a lot of extra code.

Any thoughts?

Reply:
Right, I missed that. I guess you could to a weird conversion like `From<(Scope, String)> for ScopedString`. Or maybe  use another method name, although I can't think of anything right now.